app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,28 @@
+class CommentsController < ApplicationController
+    def new
+      respond_to do |format|
+        format.html { render :new, locals: { comment: Comment.new } }
+      end
+    end
+  
+    def create
+      user = current_user
+      post = current_post
+      comment = Comment.new(comment_params)
+      comment.author = user
+      comment.post = post
+      if comment.save
+        flash[:success] = 'Comment saved successfully'
+        redirect_to user_post_url(id: post.id)
+      else
+        flash[:error] = 'Error: Comment could not be saved'
+        redirect_to new_user_post_comment_url
+      end
+    end
```

Comment:
- [ ] I couldn't create a comment. Kindly fix this function, we are only told to create a current user function and not a current post. Users should be able to create comments and posts in the way they want, not only one post.

![comm](https://user-images.githubusercontent.com/53124548/206857149-4b52b75a-954f-4902-823d-82a2f803188d.png)